ui/src/components/Gantt.js

```diff
@@ -315,14 +307,13 @@ class Gantt extends Component {
     return false;
   }
 
-  componentDidUpdate() {
-    gantt.refreshData();
-    gantt.sort("text", false);
-    gantt.render();
-    gantt.scrollTo(this.scrollPos.x, this.scrollPos.y);
-  }
-
   render() {
+    this.scrollPos = gantt.getScrollState();
+    this.expandedTasks = new Map();
+    gantt.eachTask(task => {
+      this.expandedTasks.set(task.id, task.$open);
```

Comment:
Does that work when you switch between projects?

Reply:
This `eachTask` works fine, as it only gathers data. But the part below should leave the task open if it's not in the map, let me fix it.